### ISSUE TITLE
options.mounted hook opportunity

### ISSUE THREAD

**Issue by LongTengDao** [vuejs/vuejs.org#2286]
> Note that `mounted` does **not** guarantee that all child components have also been mounted. If you want to wait until the entire view has been rendered, you can use `vm.$nextTick` inside of `mounted`

I found parent.mounted hook only called after all children.mounted called, expect the child component is async. But in that case, `vm.$nextTick` won't help too.

Is there any mistake of me or the doc?

**Comment by sarahdayan** [comment]
Hey @LongTengDao! I think the documentation is correct since the `mounted` hook triggers once the component did mount, it doesn't wait for children to mount. So, if you have a component `A` that renders a component `B`, the `mounted` hook of `A` executes before the one of `B`.

This example [illustrates it](https://codesandbox.io/s/dreamy-margulis-xzxb3).

Does this help?

**Comment by LongTengDao** [comment]
I add one console on your example (https://codesandbox.io/s/frosty-night-9o8ku), it shows:
```
Child is mounted
Parent is mounted
Child mounted nextTick
Parent mounted nextTick
```
Parent is always after Child, wether you use nextTick

**Comment by sarahdayan** [comment]
Indeed, I just realized that in my sandbox, the child `console.log` was in a `nextTick` callback 🤦‍♀

However, as you can see in your demo and mine, the parent's `nextTick` callback always runs last. So the documentation is correct when it says that using it in the parent lets you know that the full tree was rendered.

**Comment by LongTengDao** [comment]
But the "parent mounted" event is also after "child mounted" event, I can't see the difference between using "mounted" and "nextTick" (which the doc want to express)

**Comment by sarahdayan** [comment]
🧐 @sdras @NataliaTepluhina any idea?
